### PR TITLE
Xattr test

### PR DESCRIPTION
**Pull request by PaulZ-98**
This PR is only to exercise the xattr portion of the CI testing, in order to find a memory leak that the test is revealing.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Comment by behlendorf** [comment]
I'm not sure why this wasn't able to reproduce the issue.  But it would probably be worthwhile to let it run the full ZTS to try and reproduce the issue.

**Comment by PaulZ-98** [comment]
I have rebased and removed the tag limiting the test to xattr.  The full test suite will run in the CI.

**Comment by behlendorf** [comment]
It appears the full CI run was able to [reproduce the issue](http://build.zfsonlinux.org/builders/Ubuntu%2017.04%20x86_64%20Coverage%20%28TEST%29/builds/7717/steps/shell_9/logs/kmemleak).

**Comment by PaulZ-98** [comment]
I want to instrument the code with either cmn_err or zfs_dbgmsg.  How can I ensure that I get to see the messages when the CI finds the kmemleak?

**Comment by behlendorf** [comment]
You might try manually updating these test cases to scan for kmemleaks during cleanup.  Then if any are found failing the test case which will cause the dbgmsg log to be dumped and saved.  Here's the relevant bit of the kmemleak script which checks for leaks.

https://github.com/zfsonlinux/zfs-buildbot/blob/master/scripts/bb-test-zfstests.sh#L163

File: tests/zfs-tests/tests/functional/xattr/xattr_common.kshlib [hunk 1]
```diff
@@ -134,4 +134,24 @@ function create_expected_output  { # expected_output_file  contents_of_the_outpu
    do
       log_must eval "echo $line >> $FILE"
    done
- }
+}
+
+function check_mem_leaks {
+	log_note "Checking for leaks"
+	if $(sudo -E test -e "$KMEMLEAK_FILE"); then
```
Comment (behlendorf) [inline tests/zfs-tests/tests/functional/xattr/xattr_common.kshlib]: You'll probably need to adapt this a little bit for the ZTS environment, for example the sudo calls  shouldn't be needed.  And make sure to define `KMEMLEAK_FILE="/sys/kernel/debug/kmemleak"` and `KMEMLEAK_FILE` to something reasonable.

Comment (PaulZ-98) [reply tests/zfs-tests/tests/functional/xattr/xattr_common.kshlib]: Oh good point...

**Comment by PaulZ-98** [comment]
I haven't been able to root cause the leak but seems it consistently happens on Ubuntu 17.04 CI.

**Pull request closed without merging**